src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -78,8 +77,30 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
 
             // We only want to adjust the range that was affected.
             // We need to take into account the lines affected by formatting as well as cleanup.
-            var cleanupLineDelta = LineDelta(formattedText, cleanupChanges);
-            var rangeToAdjust = new Range(rangeAfterFormatting.Start, new Position(rangeAfterFormatting.End.Line + cleanupLineDelta, 0));
+            var lineDelta = LineDelta(formattedText, cleanupChanges);
+
+            // Okay hear me out, I know this looks lazy, but it totally makes sense.
+            // This method is called with edits that the C# formatter wants to make, and from those edits we work out which
+            // other edits to apply etc. Fine, all good so far. BUT its totally possible that the user typed a closing brace
+            // in the same position as the C# formatter thought it should be, on the line _after_ the code that the C# formatter
```

Comment:
Hmmm, if we were to do C# OnTypeFormatting in the Razor language server (would require the Roslyn change to expose this in Razor.ExternalAccess) would that change what we could do? Aka would be able to 0 out the indentation for the lines prior to calling C# to format or something?